docs/api-reference/react-map-gl-draw/react-map-gl-draw.md

```diff
@@ -10,14 +12,22 @@
   - `EditorModes.DRAW_POINT` - Lets you draw a GeoJson `Point` feature.
   - `EditorModes.DRAW_RECTANGLE` - Lets you draw a `Rectangle` (represented as GeoJson `Polygon` feature).
 
-- `selectedFeatureId` (String, Optional) - id of the selected feature. `EditorModes` assigns a unique id to each feature which is stored in `feature.properties.id`.
-- `clickRadius` (Number, optional) - Radius to detect features around a hovered or clicked point. Default value is `0`
-
-- `onSelect` (Function, Required) - callback when a feature is selected. Receives an object containing `selectedFeatureId`.
-- `onUpdate` (Function, Required) - callback when anything is updated. Receives one argument `features` that is the updated list of GeoJSON features.
+- `features` (Feature[], Optional) - List of features in GeoJson format.
+- `selectedFeatureIndex` (String, Optional) - Index of the selected feature. `EditorModes` assigns a unique id to each feature which is stored in `feature.properties.id`.
+- `clickRadius` (Number, Optional) - Radius to detect features around a hovered or clicked point. Default value is `0`
 
-Feature object structure:
-`react-map-gl-draw` is stateful component.
+- `onSelect` (Function, Optional) - callback when a feature or an editHandle is selected. Receives an object containing the following parameters
+  - `selectedFeatureIndex`: selected feature index.
+  - `editHandleIndex`: selected editHandle index.
```

Comment:
the main purpose is to support point (vertex) deletion and styling

Reply:
gotcha. In `ModifyMode`, vertexes are deleted by just clicking them. There's no selection required to delete. We should discuss how we want it to work. If selection is important, we can look at what it will take to implement in the `EditMode` interface.